src/python/pants/backend/python/util_rules/pex_requirements_test.py

```diff
@@ -347,3 +348,78 @@ def test_pex_lockfile_requirement_count() -> None:
         )
         == 3
     )
+
+
+class TestResolvePexConfigPexArgs:
```

Comment:
Pants doesn't usually use these sort of classes to group tests, instead of just top-level test functions. What's the thinking with this approach here?

Reply:
Agreed. This is remnant of a `unittest` style era, not all tests have been refactored to the pytest style yet, so those old ones that still exist shouldn't be used as role models :) c.f. #13208

Reply:
I'm a little confused here.  Python is an object oriented language, Pants does not otherwise eschew objects, and grouping tests with classes is a common idiom pytest has [right in the getting started guide](https://docs.pytest.org/en/8.0.x/getting-started.html#group-multiple-tests-in-a-class).

It's all dicts and functions somewhere down there to Python, you can totally do OOP-y stuff without "objects" as such (Hello C!), and the silicon doesn't care.  So if Pants has a style guide that has some reason for not using objects -- or no reason! that's fine too! -- I'd know how to adjust.  But in this case I did it this way because an idiomatic way to test the methods of a class is a test class with methods.

Reply:
Ah sorry, I'm not sure it's explicitly covered by https://www.pantsbuild.org/2.20/docs/contributions/development/style-guide#tests, given it falls somewhere between the two examples there (not `unittest`, but also not covered by the "Good" example), but top-level to be the implicit style. I don't think it's worth blocking this PR, but @kaos is the old timer with more of the history here so his vote is stronger 😄 

I'll wait until tomorrow and get it in if we don't hear (given how small it is).

Reply:
Ah, right. yea the current style is class-less, but I see what you mean. (at first I failed to observe this was new tests altogether..)

I'm OK with using a bare class (i.e. not a `TestCase` derived one) to group related tests under. We should however update the [style guide](https://www.pantsbuild.org/2.19/docs/contributions/development/style-guide#use-pytest-style-instead-of-unittest) accordingly to cover this regardless of outcome. 

Perhaps take a vote in #development slack?

Reply:
I'm gonna merge this. It sounds like this is perfectly fine for pytest, and it's a bit silly to block a PR on the basis of something small like this.

@kaos I don't feel strongly enough about this to spend my energy defining the style here. I assume you will do that if you want to? But, if you do so and the result is that we need to come back and change this, I'm happy to do it (since I'm the one doing the merge).